tests/rptest/scale_tests/audit_log_test.py

```diff
@@ -0,0 +1,412 @@
+            'audit_log_replication_factor':
+            3,
+
+            # Enable auditing for every supported event, there are currently 8
+            'audit_enabled_event_types': [
+                'management', 'produce', 'consume', 'describe', 'heartbeat',
+                'authenticate', 'admin', 'schema_registry'
+            ],
+
+            # Adjust this value to give the system a chance to buffer more data
+            # at the kafka client size
+            # default: 16MiB
+            'audit_client_max_buffer_size':
+            16000000,
+
+            # This is the frequency at which queues are drained for data to be
+            # buffered to the kafka client. Increase this value to buffer more
+            # data which increases the likelihood of compressing duplicate events
+            # and reducing the total amount of data sent to the audit topic
+            # default: 500ms
+            'audit_queue_drain_interval_ms':
+            2000,
+
+            # How large the per shard audit buffers are. The more unique events
+            # and the more time the data spends residing in the buffer, the
+            # larger this may have to become. If this queue is exhausted events
+            # for which audited is enabled for will fail is the event cannot be
+            # enqueued
+            # default: 1MiB
+            'audit_queue_max_buffer_size_per_shard':
+            1000000,
+        }
+
+        super().__init__(test_context=ctx,
+                         security=self.security,
+                         *args,
+                         **kwargs)
+
+        self.superuser = self.redpanda.SUPERUSER_CREDENTIALS
+        self.user_creds = AuditLogTestSecurityConfig.sasl_credentials
+
+        self._ctx = ctx
+        self.admin = Admin(self.redpanda,
+                           auth=(self.superuser.username,
+                                 self.superuser.password))
+        self._super_rpk = RpkTool(self.redpanda,
+                                  username=self.superuser.username,
+                                  password=self.superuser.password,
+                                  sasl_mechanism=self.superuser.algorithm)
+        self._user_rpk = RpkTool(self.redpanda,
+                                 username=self.user_creds.username,
+                                 password=self.user_creds.password,
+                                 sasl_mechanism=self.user_creds.algorithm)
+
+    def _create_audit_test_sasl_user(self):
+        self._super_rpk.sasl_create_user(self.user_creds.username,
+                                         self.user_creds.password)
+        self._super_rpk.sasl_allow_principal(
+            f'User:{self.user_creds.username}', ['all'], 'topic', '*')
+        self._super_rpk.sasl_allow_principal(
+            f'User:{self.user_creds.username}', ['all'], 'group', '*')
+        # Attempt to list topics to check if the above worked
+        self._user_rpk.list_topics()
```

Comment:
nit: This does check that the user is created, but doesn't verify that the user has access to the topics we want them to (e.g. if no ACLs provided for the user, this would return nothing). Not sure if that's a problem or not, just pointing it out.